SharedPackages/BrowserServicesKit/Sources/Subscription/Managers/V1toV2Bridging/SubscriptionAuthV1toV2Bridge.swift

```diff
@@ -106,12 +101,13 @@ extension DefaultSubscriptionManager: SubscriptionAuthV1toV2Bridge {
 
     /// Checks if the user is eligible for a free trial.
     ///
-    /// - Important: This method is part of a temporary bridge for the AuthV1 to AuthV2 migration.
-    ///   Once the migration to AuthV2 is complete, callers should ideally access the `storePurchaseManager()`
-    ///   on an instance of `DefaultSubscriptionManagerV2` (or the final AuthV2 manager) and then call
-    ///   `storePurchaseManager().isUserEligibleForFreeTrial()` directly.
+    /// Returns `true` for Stripe-based purchases
+    /// or delegates to the store purchase manager for App Store purchases.
     public func isUserEligibleForFreeTrial() -> Bool {
-        guard currentEnvironment.purchasePlatform != .stripe, #available(macOS 12.0, *) else { return false }
+        guard #available(macOS 12.0, *) else { return false }
+        if currentEnvironment.purchasePlatform == .stripe {
+            return true
+        }
```

Comment:
This is technically not correct, as if user is on macOS below 12.0 and `purchasePlatform` is `.stripe` then they are eligible. I think the `#available(macOS 12.0, *)` check should be paired with `storePurchaseManager().isUserEligibleForFreeTrial()`

Reply:
Thanks, addressed. Please review changes.